### DISCUSSION TITLE
Interface failure

### DISCUSSION THREAD

**Discussion by deepakbiitk** [peridigm/peridigm#334] [General]
Hi all,

In a bimaterial system, I created two blocks assigned different material properties and also  different critical stretches for each block. The model runs. I would like to assign different critical stretch for bonds at the interface. Is there a way to do so? 

Thanks,
Deepak

**Comment by djlittl** [comment]
This feature isn't implemented per se, but it isn't too hard to get it to work.  I think the easiest way to do this is to store the block_id for each node as a state variable, and then modify the critical stretch code to examine the block_id for the two nodes at the end of a given bond.  You can hard code a special response for the case where the block_ids are different (which indicates an interface).  I did this once to model failure on grain boundaries, see slide 14 in https://www.osti.gov/servlets/purl/1115941

**Reply by deepakbiitk** [reply]
Thank you Dr. Littlewood for you suggestion. I appreciate for sharing the slides.
